.github/workflows/manual-publish.yml

```diff
@@ -34,3 +36,17 @@ jobs:
         uses: ./.github/actions/publish
         with:
           dry_run: ${{ inputs.dry_run }}
+
+  release-provenance:
+    needs: [ 'build-publish' ]
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.7.0
+    with:
+      base64-subjects: "${{ needs.build-publish.outputs.gem-hash }}"
+      upload-assets: true
+      upload-tag-name: TBD
```

Comment:
Same comments here as the python one. upload-assets might be good to tie into the `dry_run` flag. And TBD needs TBDed.